Source/WebInspectorUI/UserInterface/Views/DOMTreeOutline.js

```diff
@@ -86,6 +86,9 @@ WI.DOMTreeOutline = class DOMTreeOutline extends WI.TreeOutline
     setVisible(visible, omitFocus)
     {
         this._visible = visible;
+
+        this.dispatchEventToListeners(WI.DOMTreeOutline.Event.VisibilityChanged, {visible});
```

Comment:
i wonder if instead of just listening for this specific case we could/should instead listen for something more generic like `WI.ContentBrowser.Event.CurrentContentViewDidChange` and/or `WI.ContentViewContainer.Event.CurrentContentViewDidChange` 🤔

i wouldn't be surprised if there's other things like this elsewhere, so that would fix all of them (and future ones) in one go

Reply:
> i wonder if instead of just listening for this specific case we could/should instead listen for something more generic

Do you mean adding something like this to `WI.Popover`?

```js
WI.tabBrowser.contentViewContainer.addEventListener("WI.ContentViewContainer.Event.CurrentContentViewDidChange", () => { 
    this.dismiss(); 
}, this);
```

(Assume the `contentViewContainer` getter exists on `WI.TabBrowser`)

This would solve the issue for all `WI.Popover` instances and make them self-dismiss.
But it makes `WI.Popover`  know about `WI.tabBrowser`. Is that ok?

Reply:
> hmmm oddly i dont see this behavior when clicking on a source code location link in the WI.Popover 🤔
> and even weirder, once i've done that then i cant reproduce this issue

Indeed, clicking the link to jump to source doesn't exhibit the issue.
And yes, once that steps is done, jumping to the breakpoint also doesn't reproduce the issue anymore.

I thought it had something to do with this failed assertion. 
But the assertion is tripped in both scenarios: jump-to-breakpoint _and_  jump-to-source, then jump-to-breakpoint 🤔 

<img width="923" alt="Screenshot 2023-04-22 at 14 21 37" src="https://user-images.githubusercontent.com/105809798/233784502-af71d6fb-38ab-4ef8-b44b-bb720958d5f5.png">

Reply:
id love for us to understand why this is different and maybe integrate that with our solution